src/transform/mod.rs

```diff
@@ -169,13 +181,21 @@ pub fn map_descriptions(ir: &mut IR, mut ff: impl FnMut(&str) -> String) -> anyh
     Ok(())
 }
 
-fn remap_names<T>(x: &mut HashMap<String, T>, f: impl Fn(&mut String)) {
+fn remap_names<T>(
+    x: &mut HashMap<String, T>,
+    f: impl Fn(&mut String),
+) -> Result<(), (String, String)> {
```

Comment:
This should return a custom error with Debug/Display impls, this way you can do `.unwrap()` in the above calls instead of copying the error message 4 times.